File: examples/node/generate-text.mjs
/**
 * Generate text using Sats4AI L402 API.
 *
 * Usage:
 *   node generate-text.mjs "Write a haiku about Bitcoin"
 *
 * Requires: Node.js 18+ (built-in fetch)
 */

import { requestL402, completeL402 } from "../../lib/l402-client.mjs";
import { createInterface } from "readline";

const message = process.argv[2] || "Write a haiku about Bitcoin";

console.log(`Requesting text: '${message}'`);

const body = { input: [{ role: "user", content: message }] };

// Step 1: Request → 402
const challenge = await requestL402("generate-text", body);

if (challenge.alreadyPaid) {
  console.log(challenge.result?.body || challenge.result);
  process.exit(0);
}

// Step 2: Pay the invoice
console.log(`\nInvoice to pay:\n${challenge.invoice}\n`);

const rl = createInterface({ input: process.stdin, output: process.stdout });
const preimage = await new Promise((resolve) =>
  rl.question("Preimage (hex): ", (answer) => {
    rl.close();
    resolve(answer.trim());
  })
);

// Step 3: Submit with proof
console.log("\nSubmitting with L402 proof...");
const result = await completeL402(challenge, body, preimage);
console.log(`\n${result?.body || JSON.stringify(result)}`);

File: examples/python/generate_text.py
"""
Generate text using Sats4AI L402 API.

Usage:
    python generate_text.py "Write a haiku about Bitcoin"

Requires: pip install requests
"""

import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../.."))
from lib.l402_client import L402Client

message = sys.argv[1] if len(sys.argv) > 1 else "Write a haiku about Bitcoin"

client = L402Client()

# Step 1: Request → 402
print(f"Requesting text: '{message}'")
challenge = client.request("generate-text", {
    "input": [{"role": "user", "content": message}],
})

if challenge.get("already_paid"):
    print(challenge["result"].get("body", challenge["result"]))
    sys.exit(0)

# Step 2: Pay the invoice
print(f"\nInvoice to pay:\n{challenge['invoice']}\n")
preimage = input("Preimage (hex): ").strip()

# Step 3: Submit with proof
print("\nSubmitting with L402 proof...")
result = client.complete(challenge, preimage)

if isinstance(result, dict):
    print(f"\n{result.get('body', result)}")
else:
    print(f"\n{result}")
